onnxruntime/core/graph/function_utils.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/common.h"
+#include "onnx/onnx_pb.h"
+#include "core/graph/graph.h"
+#include "core/graph/indexed_sub_graph.h"
+#include "core/graph/function.h"
+#include "core/graph/schema_registry.h"
+
+namespace onnxruntime {
+namespace function_utils {
+
+std::unique_ptr<ONNX_NAMESPACE::OpSchema> CreateSchema(const Graph& graph,
+                                                       const IndexedSubGraph& nodes_to_fuse);
+
+std::unique_ptr<ONNX_NAMESPACE::OpSchema> CreateSchema(const std::string& function_domain,
+                                                       const std::string& function_name,
+                                                       const std::unordered_map<std::string, const ONNX_NAMESPACE::FunctionProto*>& model_local_functions,
+                                                       const std::unordered_map<std::string, int>& domain_version_map,
+                                                       const SchemaRegistryManager& schema_registry,
+                                                       const logging::Logger& logger,
+                                                       bool allow_released_opsets_only);
+
+/** Get the unique id for function. This is used as a key to find the 
+* relevant model local function from it's container.
+* @param function_domain Domain for the function.
+* @param function_name Name of the function. Name should match the OpType of the node which references the function.
+*/
+inline std::string GetFunctionIdentifier(const std::string& function_domain, const std::string& function_name) {
+  return function_domain + ":" + function_name;
```

Comment:
>function_domain [](http://example.com/codeflow?start=9&length=15)

What if function_domain contains ":" ? Is there any thing to prevent it?

Reply:
Opps, accidently click "resolve" on another comments.... lol 
Let me put response together here.

This "GetFunctionIdentifier" is inherit from existing implementation, like i mentioned, i just re-organized the code. So i don't have enough context to answer your question, like did we consider if function domains contains ":" and whether onnx prevent it. @askhade and @gramalingam may have better context for that.

I agree with your suggestion like we can provide a type and override operator== or hash function. For the key generation here, even func domain contains ":", the generated id seems still unique?

Reply:
> even func domain contains ":", the generated id seems still unique

Not really.  Consider "a:b" + ":" +"c" vs "a" + ":b:c"

Reply:
ok, you are right. @gramalingam and @askhade , do we have andy validation in onnx?

Reply:
Technically, the ONNX standard says domain should be like "com.microsoft", valid identifiers with dots. I don't think ":" is legal. Similarly, the optype is expected to be a valid C identifier. 

Having said that, it is a good safety measure to define a POD type as discussed. The identifier restriction is routinely violated when naming tensors, with names like "1" and "2", which is one reason why identifier-checking was not made strict in ONNX. (But I haven't seen such violations in naming ops or domains.)